packages/automerge-repo-storage-indexeddb/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import { StorageAdapter } from "@automerge/automerge-repo"
+
+export class IndexedDBStorageAdapter extends StorageAdapter {
+  private dbPromise: Promise<IDBDatabase>
+
+  constructor() {
+    super()
+    this.dbPromise = new Promise((resolve, reject) => {
+      const request = indexedDB.open("automerge", 1)
+
+      request.onerror = function () {
+        reject("Database error: " + request.error)
+      }
+
+      request.onupgradeneeded = function (event: IDBVersionChangeEvent) {
+        const db = (event.target as IDBOpenDBRequest).result
+        db.createObjectStore("documents") // No keyPath specified here.
+      }
+
+      request.onsuccess = function (event) {
+        const db = (event.target as IDBOpenDBRequest).result as IDBDatabase
+        resolve(db)
+      }
+    })
+  }
+
+  async load(keyArray: string[]): Promise<Uint8Array | undefined> {
+    const db = await this.dbPromise
+
+    return new Promise((resolve, reject) => {
+      const transaction = db.transaction(["documents"])
+      const objectStore = transaction.objectStore("documents")
+      const request = objectStore.get(keyArray)
+
+      request.onerror = function () {
+        reject("Unable to retrieve data from database!")
+      }
+
+      request.onsuccess = function (event) {
+        resolve(
+          ((event.target as IDBRequest).result as { binary: Uint8Array })
+            ?.binary
+        )
+      }
+    })
+  }
+
+  async save(keyArray: string[], binary: Uint8Array): Promise<void> {
+    const db = await this.dbPromise
+    return new Promise((resolve, reject) => {
+      const transaction = db.transaction(["documents"], "readwrite")
+      const objectStore = transaction.objectStore("documents")
+      const request = objectStore.put(
+        { key: keyArray, binary: binary },
+        keyArray
+      )
+
+      request.onerror = function () {
+        reject("Unable to save data to database!")
```

Comment:
please include the error object (so we can tell why it errored)

Reply:
fair enough. did 'em all.